staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -576,6 +576,14 @@ const (
 	// Status code 500
 	StatusReasonServerTimeout StatusReason = "ServerTimeout"
 
+	// StatusReasonInitializationTimeout means the object is created but server has
+	// timed out waiting for its initialization
+	// Details (optional):
+	//   "kind" string - the kind attribute of the resource being acted on.
+	//   "id"   string - the operation that is being attempted.
+	// Status code 500
+	StatusReasonInitializationTimeout StatusReason = "InitializationTimeout"
```

Comment:
I don't think we need to have a new reason.  Just add a new `cause`.  Cause was intended for this.